### DISCUSSION TITLE
ignoreNpmrcFile is missing from the docs

### DISCUSSION THREAD

**Discussion by anopheles** [renovatebot/renovate#11304] [Suggest an Idea]
I can't seem to find the option `ignoreNpmrcFile` documented in https://docs.renovatebot.com/

Is this intentional?

**Comment by viceice** [comment]
Yes, was removed. checkout https://docs.renovatebot.com/getting-started/private-packages/#npm

**Reply by anopheles** [reply]
The only thing that I can see that the above doc does not contain the string "ignoreNpmrcFile". I can't seem to find a different way to tell renovate to behave similar to the ignoreNpmrcFile flag. 

Furthermore renovate still uses the flag, see https://github.com/renovatebot/renovate/blob/569bee6e7d78956bac61831794ba54272f314e42/lib/config/migration.ts#L210

**Reply by anopheles** [reply]
I'm not sure what the multiple 👎 are trying to tell me. Was this a bad question to ask? Isn't there a migration strategy in place? Shouldn't we use this flag anymore? If yes, why is there no warning in the output when executing renovate?

**Reply by rarkins** [reply]
I gave your comment a thumbs down for its snarky tone combined with the fact that the page you were linked to includes this:

![image](https://user-images.githubusercontent.com/6311784/129794937-f5a4ba41-50e5-4b1f-9d24-99f9243907bf.png)

You can also see from the source you linked to that if `ignoreNpmrcFile` is included in config then Renovate migrates it to be an empty `npmrc` string, which achieves the same thing. I don't know why you're still asking "Isn't there a migration strategy in place" after literally linking to a file called `migration.ts` immediately above it.

The removal of `ignoreNpmrc` was made in v25.0.0 and documented: https://github.com/renovatebot/renovate/releases/tag/25.0.0

If you upgraded from <25 without reading the breaking changes, that's on you. If you are new to Renovate since v25 and followed out of date documentation here, that's on us - but that seems to be the opposite of what you're complaining about.

Your original question was "Is this intentional" and the answer you received was Yes, it was intentional. Adding a link to a page which includes more information was really a bonus. If someone answers you in 14 minutes in an open source project you should ideally be grateful, but definitely not rude. Please be nicer in future if you'd like to keep receiving support here.

**Reply by anopheles** [reply]
Sorry If my reply came off as rude. That was not my intention at all! But yes, after reading it again I can see that this might be the case. My intention was to provide others with guidance that may run into the same problem that I encountered while I was integrating renovate into our repository.

> I don't know why you're still asking "Isn't there a migration strategy in place" after literally linking to a file called migration.ts immediately above it.

I don't know the internals of renovate so I can only assume what `migration.ts` does. My bad for not digging deeper though. I did not assume that it is responsible for migrating deprecated configuration flags.

> The removal of ignoreNpmrc was made in v25.0.0 and documented

Thanks. Here it clearly says 
> npm: ignoreNpmrcFile is no longer supported. Use an empty string for config.npmrc instead.

That's what I was looking for! Thanks you very much. 

I definitely looked over this part in the documentation:

> If configured like this, Renovate will use this to authenticate with npm and will ignore any .npmrc files(s) it finds checked into the repository.

> If you upgraded from <25 without reading the breaking changes, that's on you.

I am a first time user of renovate. I copied above flag from an old renovate issue that solved my problem that I had and I was unable to solve otherwise: renovate was always taking the .npmrc file from the repository (which had an unusable configuration). I wanted to use my globally defined one.

> Please be nicer in future if you'd like to keep receiving support here.

Sorry for the noise. I just wanted to be on point. Again, thanks for your support, I really appreciate it.

**Reply by anopheles** [reply]
In my defense, the section under

> Add npmrc string to Renovate config

in https://docs.renovatebot.com/getting-started/private-packages/#npm

does not say what behavior we can expect if we leave it empty. IMHO the documentation could be clearer here.
